src/v/pandaproxy/rest/handlers.cc

```diff
@@ -72,18 +73,44 @@ ss::shard_id consumer_shard(const kafka::group_id& g_id) {
 
 ss::future<server::reply_t>
 get_brokers(server::request_t rq, server::reply_t rp) {
+    ss::sharded<kafka::client::client> proxy_client;
+    ss::sstring username{};
+    ss::sstring userpass{};
+
+    if (rq.service().config().rest_require_auth()) {
+        auto creds_pair = parse::basic_auth_header(*rq.req);
+        username = creds_pair.first;
+        userpass = creds_pair.second;
+    }
+
     auto res_fmt = parse::accept_header(
       *rq.req,
       {json::serialization_format::v2, json::serialization_format::none});
     rq.req.reset();
     vlog(plog.debug, "get_brokers");
 
+    // Process request after reset
+    if (rq.service().config().rest_require_auth()) {
+        // Create new kafka client config with scram creds
+        // and use sha256 by default
+        kafka::client::configuration proxy_client_config;
+        proxy_client_config.sasl_mechanism.set_value(
+          ss::sstring{"SCRAM-SHA-256"});
+        proxy_client_config.scram_username.set_value(username);
+        proxy_client_config.scram_password.set_value(userpass);
+
+        auto client_cfg_as_yml = to_yaml(
+          proxy_client_config, config::redact_secrets::no);
+        auto f = proxy_client.start(client_cfg_as_yml);
+        rq.deferred.emplace_back(
+          [&proxy_client] { proxy_client.stop().get(); });
+        f.get();
```

Comment:
the problem is because the client fails to start so this f.get also causes a crash... I feel this the solution should be obvious but I'm missing it.

Reply:
I think `.get()` isn't valid since this isn't in an `ss::thread`.

Reply:
I'm doing something different now.